package/MDAnalysis/core/groups.py

```diff
@@ -1039,32 +1039,58 @@ def wrap(self, compound="atoms", center="com", box=None):
             if not all(s == 0.0):
                 o.atoms.translate(s)
 
-    def groupby(self, topattr):
+    def groupby(self, *topattrs):
         """Group together items in this group according to values of *topattr*
 
         Parameters
         ----------
         topattr: str
-           Topology attribute to group components by.
+           One or more topology attribute to group components by.
 
         Returns
         -------
         dict
-            Unique values of the topology attribute as keys, Groups as values.
+            Unique values of the multiple combinations of topology attributes 
+            as keys, Groups as values.
 
         Example
         -------
-        To group atoms with the same mass together::
-
-          >>> ag.groupby('masses')
-          {12.010999999999999: <AtomGroup with 462 atoms>,
-          14.007: <AtomGroup with 116 atoms>,
-          15.999000000000001: <AtomGroup with 134 atoms>}
-
+       
+        
+        To group atoms with the same mass together::     
+                
+        >>> ag.groupby('masses')
+        {12.010999999999999: <AtomGroup with 462 atoms>,
+         14.007: <AtomGroup with 116 atoms>,
+         15.999000000000001: <AtomGroup with 134 atoms>}        
+
+        To group atoms with the same residue name and mass together::
+
+          >>> ag.groupby('resnames', 'masses')
```

Comment:
About the API. It might be better to follow the pandas/datreant examples and use an explicit list for multiple arguments. Then our groupby API is as close as possible to others minimizing surprises for the user.

```python
# MDAnalysis
ag.groupby(['masses', 'charges'])
# pandas
df.groupby(['columnname', 'othercolumn'])
# datreant
bundle.categories.groupby(['category1', 'categorie2'])
```